source/Xunit.Gherkin.Quick/ParameterHelper.cs

```diff
@@ -0,0 +1,44 @@
+using Gherkin.Ast;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace Xunit.Gherkin.Quick
+{
+    internal static class ParameterHelper
+    {
+        public static List<object> GetParamValues(MethodInfo method, Step step, List<string> specificationParameters)
```

Comment:
There is a complex logic inside this. I hope you smoke tested this thoroughly? If this works fine with different cases (such as param at the end or inside), then should be fine. Just want to avoid defects leaking.

Reply:
I've actually written a small test of tests for this, but as unit testing style is something theres often disagreement on I've left it out of this PR until the other refactor is all merged in. I've just pushed my working branch with the unit tests to here if you want to take a look?
https://github.com/ttutisani/Xunit.Gherkin.Quick/tree/TableDataSupportWithUnitTests

Reply:
In fact, I've just noticed the refactor has been merged, I'll include the tests in here for this one (sorry, working top down haha!)